types/parse-prefer-header/index.d.ts

```diff
@@ -1,8 +1,8 @@
 // Type definitions for parse-prefer-header 1.0
 // Project: https://github.com/ppaskaris/node-parse-prefer-header
-// Definitions by: Vincenzo Chianese <https://github.com/XVincentX>
+// Definitions by: Vincenzo Chianese <https://github.com/XVincentX>, Marcell Toth <https://github.com/marcelltoth>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
-declare function parsePreferHeader(preferHeader: string | string[]): { [key: string]: string | boolean };
+declare function parsePreferHeader(preferHeader: string | ReadonlyArray<string> | null | undefined): { [key: string]: string | true };
```

Comment:
`ReadonlyArray` is not very convenient since it will force 99% of the audience (not using it) to make a cast or something else in order to use this function

Reply:
It will not. `ReadonlyArray` is a subset of `Array`.